examples/templates/winapi_threads.h

```diff
@@ -0,0 +1,22 @@
+#ifndef __WINAPI_THREAD_TEMPLATE_H__
+#define __WINAPI_THREAD_TEMPLATE_H__
+
+#include <process.h>
+
+typedef uintptr_t thread_handle_type;
+typedef void thread_return_type;
+
+int create_thread(thread_handle_type* thread, thread_return_type(*start_routine) (void *), void *arg)
+{
+    uintptr_t handle = _beginthread(start_routine, 0, arg);
+    *thread = handle;
+    return handle ? 0 : 1;
+}
+
+void cancel_thread(thread_handle_type handle)
+{
+    /* Bad idea: */
+    TerminateThread((HANDLE)handle, 0);
```

Comment:
Terminating a thread externally is always a bad idea, it would be cleaner to signal the thread that it should exit.
I kept the winapi implementation as close as possible to the posix one.

Reply:
I agree. The examples are meant to be as simple as possible.